rocky/rocky.go

```diff
@@ -137,53 +82,60 @@ func NewConfig(opts ...option) Config {
 }
 
 func (c Config) Update() error {
-	// "8" is an alias of the latest release that doesn't contain old security advisories,
-	// so we have to get all available minor releases like 8.5 and 8.6 so that we can have all the advisories.
-	releases, err := c.getReleasesList()
-	if err != nil {
-		return xerrors.Errorf("failed to get a list of Rocky Linux releases: %w", err)
-	}
-	for _, release := range releases {
-		for _, repo := range c.repos {
-			for _, arch := range c.arches {
-				log.Printf("Fetching Rocky Linux %s %s %s data...", release, repo, arch)
-				if err = c.update(release, repo, arch); err != nil {
-					return xerrors.Errorf("failed to update security advisories of Rocky Linux %s %s %s: %w", release, repo, arch, err)
+	// there are 2 different urls for actual and old releases
+	for _, baseUrl := range c.baseUrls {
+		errataWasWrite := false
+		// "8" is an alias of the latest release that doesn't contain old security advisories,
+		// so we have to get all available minor releases like 8.5 and 8.6 so that we can have all the advisories.
+		releases, err := c.getReleasesList(baseUrl)
+		if err != nil {
+			return xerrors.Errorf("failed to get a list of Rocky Linux releases: %w", err)
+		}
+		for _, release := range releases {
+			for _, repo := range c.repos {
+				for _, arch := range c.arches {
+					log.Printf("Fetching Rocky Linux %s %s %s data...", release, repo, arch)
+					wasWrite, err := c.update(release, repo, arch, baseUrl)
+					if err != nil {
+						return xerrors.Errorf("failed to update security advisories of Rocky Linux %s %s %s: %w", release, repo, arch, err)
+					}
+					if wasWrite {
+						errataWasWrite = true
+					}
 				}
 			}
 		}
+		// we should know if url doesn't contain vulnerabilities
+		// probably database has been changed
+		if !errataWasWrite {
```

Comment:
What is this `errataWasWrite` for? Can't we just return an error?

Reply:
https://dl.rockylinux.org/vault/rocky has [8.3](http://dl.rockylinux.org/vault/rocky/8.3) and [8.4-RC1](http://dl.rockylinux.org/vault/rocky/8.4-RC1) folders without `updateinfo` field in `repomd.xml`.
https://download.rockylinux.org/pub/rocky has [8.4](http://download.rockylinux.org/pub/rocky/8.4/) and [8.5](http://download.rockylinux.org/pub/rocky/8.5/) folders without `repomd.xml` files.
These folders are returning errors, and I've added skip these folders.
But, for example, if database is moved to another link, we don't see that.
To do this, i added this check: if link (https://dl.rockylinux.org/vault/rocky or https://download.rockylinux.org/pub/rocky) doesn't have advisories - return error.

Reply:
Got it. Looks like `update` doesn't need to return bool. I refactored a bit. https://github.com/aquasecurity/vuln-list-update/pull/161/commits/9fb03250ccfcaf97f4f9a55ea049ff63affd63d6